src/EchoServer.java

```diff
@@ -0,0 +1,51 @@
+public class EchoServer {
+
+    public static final String PROMPT = "Type something ('exit' to quit)";
+    private Console console;
+
+    public EchoServer(Console console) {
+        this.console = console;
+    }
+
+    public static void main(String[] args) {
+        EchoServer echoServer = new EchoServer(new EchoConsole(System.in, System.out));
+        echoServer.run();
+    }
+
+    public void run() {
+        String message = "";
+
+        do {
+            message = printUserInput();
+        } while (!userTypedExit(message));
```

Comment:
Try to avoid negation in your conditions. Can you think of a better name for the affirmative?